drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -378,6 +382,45 @@ static int rf2xx_filter(struct device *dev,
 	return -ENOTSUP;
 }
 
+#if !defined(CONFIG_IEEE802154_RAW_MODE) && \
+	defined(CONFIG_NET_L2_OPENTHREAD)
+
+#define ACK_FRAME_LEN 3
+#define ACK_FRAME_TYPE (2 << 0)
+#define ACK_FRAME_PENDING_BIT (1 << 4)
+
+static void rf2xx_handle_ack(struct rf2xx_context *ctx, u8_t seq_number)
+{
+	struct net_pkt *pkt;
+	u8_t ack_psdu[ACK_FRAME_LEN] = {ACK_FRAME_TYPE, 0, seq_number};
+
+	pkt = net_pkt_alloc_with_buffer(ctx->iface, ACK_FRAME_LEN,
```

Comment:
What about allocating pkt on stack? it's a bit heavy yes, but at least you won't have to call the allocator, and thus no allocation error to handle (note that ieee802154_radio_handle_ack() do not mangle with pkt ref so it will be fine here)

Reply:
Just something like:
struct net_buf buf = {
    .data = ack_psdu,
    .len = ACK_FRAME_LEN,
    .size = ACK_FRAME_LEN
};
struct net_pkt pkt = { .buffer = &buf };

and there you go.